api/src/test/java/org/pmiops/workbench/api/WorkspacesControllerTest.java

```diff
@@ -2318,4 +2324,179 @@ public void testGetWorkspacesDoesNotGetsPublishedIfReader() {
 
     assertThat(workspacesController.getWorkspaces().getBody().getItems().size()).isEqualTo(0);
   }
+
+  private void assertNotebookLockingMetadata(
+      Map<String, String> gcsMetadata,
+      NotebookLockingMetadataResponse expectedResponse,
+      WorkspaceACL acl) {
+
+    final String testWorkspaceNamespace = "test-ns";
+    final String testWorkspaceName = "test-ws";
+    final String testNotebook = "test-notebook.ipynb";
+
+    org.pmiops.workbench.firecloud.model.Workspace fcWorkspace =
+        createFcWorkspace(testWorkspaceNamespace, testWorkspaceName, LOGGED_IN_USER_EMAIL);
+    fcWorkspace.setBucketName(BUCKET_NAME);
+    stubGetWorkspace(fcWorkspace, WorkspaceAccessLevel.OWNER);
+    stubFcGetWorkspaceACL(acl);
+
+    final String testNotebookPath = "notebooks/" + testNotebook;
+    doReturn(gcsMetadata).when(cloudStorageService).getMetadata(BUCKET_NAME, testNotebookPath);
+
+    // this is what I want to do, but it's broken somehow (never fails)
+    //    assertThat(
+    //        workspacesController
+    //            .notebookLockingMetadata(testWorkspaceNamespace, testWorkspaceName, testNotebook)
+    //            .getBody()
+    //            .equals(expectedResponse));
+
+    NotebookLockingMetadataResponse actualResponse =
+        workspacesController
+            .notebookLockingMetadata(testWorkspaceNamespace, testWorkspaceName, testNotebook)
+            .getBody();
+
+    assertEquals(expectedResponse, actualResponse);
+  }
+
+  @Test
+  public void testNotebookLockingMetadata() {
+    final String lastLockedUser = LOGGED_IN_USER_EMAIL;
+    final Long lockExpirationTime = Instant.now().plus(Duration.ofMinutes(1)).toEpochMilli();
+
+    final Map<String, String> gcsMetadata =
+        new HashMap<String, String>() {
+          {
+            put("lockExpirationTime", lockExpirationTime.toString());
+            put(
+                "lastLockedBy",
+                WorkspacesController.notebookLockingEmailHash(BUCKET_NAME, lastLockedUser));
```

Comment:
opt: Hardcoding the current output of this function might be preferable, so you can see a concrete input rather than depending on the implementation within the test. This can help in sanity checking, e.g. if Welder was returning the hash with capital hex letters and this was expected lowercase.

Reply:
Agree that this is useful.  I added notebookLockingEmailHashTest() as an equivalent